src/basic/unit-name.c

```diff
@@ -246,11 +246,7 @@ int unit_name_build(const char *prefix, const char *instance, const char *suffix
         assert(prefix);
         assert(suffix);
         assert(ret);
-
-        if (isempty(suffix))
-                return -EINVAL;
-        if (suffix[0] != '.')
-                return -EINVAL;
+        assert(suffix[0] == '.');
```

Comment:
Asserts get compiled out if built with `-DNDEBUG` (see [docs](https://www.gnu.org/software/libc/manual/html_node/Consistency-Checking.html) and [header file](https://sourceware.org/git/?p=glibc.git;a=blob;f=assert/assert.h;h=5468c16c2c4e354e86454386c39a049daa130523;hb=23158b08a0908f381459f273a984c6fd328363cb#l43).)

Building with `-DNDEBUG` is a common setting, often pushed through CFLAGS in some build environments.

I agree with what @poettering said in the other thread: "place assert()s to catch programming errors while catching invalid data errors with runtime errors. I'd claim that a parameter being NULL where it shouldn't is really a programming error, while looking into a string could be an input data error, and hence should be handled more defensively..."